MapboxDirections/MBDirections.swift

```diff
@@ -229,11 +245,26 @@ open class Directions: NSObject {
      */
     @objc(URLForCalculatingDirectionsWithOptions:)
     open func url(forCalculating options: RouteOptions) -> URL {
-        let params = options.params + [
-            URLQueryItem(name: "access_token", value: accessToken),
-        ]
-        
-        let unparameterizedURL = URL(string: options.path, relativeTo: apiEndpoint)!
+        let isMapboxHost = apiEndpoint.host == mapboxHost
```

Comment:
This is tricky. We use access tokens on other hosts that are not equal to `mapboxHost`. Perhaps we can make `mapboxHost` public so we can override this when using other mapbox hostnames.

Reply:
As you wish. The goal here is to know when we are using mapbox host.

Reply:
The `access_token` parameter is unnecessary when talking to an OSRM endpoint, but it should already be easy enough to set the access token to something bogus, as the unit tests do. If the OSRM endpoint is erroring out because of that parameter, I guess we could’ve made `Credentials(accessToken:)` take an empty string as a signal to not add the `access_token` parameter to the URL. 🤷‍♂️